meteor/client/ui/Settings/ShowStyle/VariantSettings.tsx

```diff
@@ -43,9 +71,56 @@ export const ShowStyleVariantsSettings = withTranslation()(
 			this.state = {
 				editedMappings: [],
 				timestampedFileKey: Date.now(),
+				dndVariants: this.getOrderedShowStyleVariants(),
 			}
 		}
 
+		componentDidUpdate(
+			prevProps: Readonly<Translated<IShowStyleVariantsProps>>,
+			prevState: Readonly<IShowStyleVariantsSettingsState>
+		) {
+			if (this.showStyleVariantsChanged(prevState, prevProps)) {
+				timer = setTimeout(
+					() =>
+						this.setState({
+							dndVariants: this.getOrderedShowStyleVariants(),
+						}),
+					timeout
+				)
+			}
+		}
+
+		componentWillUnmount() {
+			clearTimeout(timer)
+		}
+
+		private showStyleVariantsChanged = (
```

Comment:
At the moment the first two if clauses both return true, and thus could (or perhabs should) be merged. Would lower the amount of duplicate lines ever so slightly.

``` typecript
if ((this.props.showStyleVariants.length > 0 && this.state.dndVariants.length === 0) || 
(this.props.showStyleVariants.length !== prevState.dndVariants.length)) {
	return true
}
```
Notice that i swapped the order of checked properties in the second one, to have the first property mentioned being the same in both check. Should ease readability ever so slightly.